.github/workflows/PR-build.yml

```diff
@@ -47,7 +47,7 @@ jobs:
         if: needs.changes.outputs.lint == 'true'
         uses: actions/setup-go@v4
         with:
-          go-version: ~1.19.6
+          go-version: ~1.21.1
```

Comment:
Nit: ~1.21.3
does the same thing since ~ will take latest patch but looks better since 1.21.1 has known bugs